<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lifan Liang" />

<meta name="date" content="2023-10-16" />

<title>cTWAS</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neuron_stimulation_100line_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">cTWAS</h1>
<h4 class="author">Lifan Liang</h4>
<h4 class="date">2023-10-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>neuron_stimulation_100line_eQTL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231015code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231015)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231015code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231015)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangneuronstimulation100lineeQTLtreec37ce0eb03205a390a363cacb3d40219c24d4965targetblankc37ce0ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/tree/c37ce0eb03205a390a363cacb3d40219c24d4965" target="_blank">c37ce0e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangneuronstimulation100lineeQTLtreec37ce0eb03205a390a363cacb3d40219c24d4965targetblankc37ce0ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/tree/c37ce0eb03205a390a363cacb3d40219c24d4965" target="_blank">c37ce0e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/cTWAS.Rmd</code>) and HTML
(<code>docs/cTWAS.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/c37ce0eb03205a390a363cacb3d40219c24d4965/analysis/cTWAS.Rmd" target="_blank">c37ce0e</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-25
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/6e265f9b24da02c81a5a9e2c70e0bc96e2d76927/docs/cTWAS.html" target="_blank">6e265f9</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/8c088c5eeae82acf35d2f9089e4c0f13073a5509/analysis/cTWAS.Rmd" target="_blank">8c088c5</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-25
</td>
<td>
wflow_publish(c("analysis/cTWAS.Rmd", "data/", "docs/"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/8c088c5eeae82acf35d2f9089e4c0f13073a5509/docs/cTWAS.html" target="_blank">8c088c5</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-25
</td>
<td>
wflow_publish(c("analysis/cTWAS.Rmd", "data/", "docs/"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/25add741283003399f3309007858358ac87b49da/docs/cTWAS.html" target="_blank">25add74</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/bb2003b432f8c95b3b7f8e4f7be9311ce623bf19/analysis/cTWAS.Rmd" target="_blank">bb2003b</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-17
</td>
<td>
wflow_publish(c("analysis/cTWAS.Rmd", "docs/assets/"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/bb2003b432f8c95b3b7f8e4f7be9311ce623bf19/docs/cTWAS.html" target="_blank">bb2003b</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-17
</td>
<td>
wflow_publish(c("analysis/cTWAS.Rmd", "docs/assets/"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/b92db66650a2e0b21c1ad041e21aca76a2edff85/analysis/cTWAS.Rmd" target="_blank">b92db66</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-17
</td>
<td>
wflow_publish("analysis/cTWAS.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/525812a8f71dff3f5cfd1bba83e3beee1bf8b6ab/docs/cTWAS.html" target="_blank">525812a</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/5c3af5aaa8ee2affd23b67583f2b5e1aceca9809/analysis/cTWAS.Rmd" target="_blank">5c3af5a</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
wflow_publish("analysis/cTWAS.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/bf5c5310c1c460359f1e25f9cd50b4b4449eb563/docs/cTWAS.html" target="_blank">bf5c531</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/221b59df39947326ece5b8134ede1358117ac886/analysis/cTWAS.Rmd" target="_blank">221b59d</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
wflow_publish(c("analysis/cTWAS.Rmd", "docs/assets/"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/7344aedb712ae15f3f1b85db1b868eaae6e649d7/docs/cTWAS.html" target="_blank">7344aed</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/ba8f9f4d1b2f2a76aa1c582782e7b3fd4bca17fb/analysis/cTWAS.Rmd" target="_blank">ba8f9f4</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
wflow_publish("analysis/cTWAS.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/af82042d6b0372220f6f0e1f879ab2617d7dc903/docs/cTWAS.html" target="_blank">af82042</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/172e4215aa553db890c8d094ac07d1913d72472d/analysis/cTWAS.Rmd" target="_blank">172e421</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
test PDF
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/2baca5f7d4cca251fb968c599f55d260cbaca2d4/docs/cTWAS.html" target="_blank">2baca5f</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/40d743dec25c4051bcaeddf6c498c7688b79c979/analysis/cTWAS.Rmd" target="_blank">40d743d</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Test run
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/neuron_stimulation_100line_eQTL/06a48c4591bfe91bdfcba82ea303f7365a024a8c/docs/cTWAS.html" target="_blank">06a48c4</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/0fc1e542a6bcbc64dacec09795186ce044c2b33a/analysis/cTWAS.Rmd" target="_blank">0fc1e54</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-16
</td>
<td>
Test run
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<p>We have run the multigroup branch of <a
href="https://github.com/xinhe-lab/ctwas">cTWAS</a> on each cell type
seperately. Three time points serve as the “group” for the cell type
run.</p>
<p>eGenes for each condition (the combination of cell type and time
point) was selected with three criteria: 1. At least one SNP-gene pair
has FDR&lt;0.05 in any time point within the same cell type; 2. P value
&lt; 0.1 for the current condition. The number of genes for different
settings were shown below; 3. Must be a protein coding gene</p>
<p>Below are the eGenes selected for imputation in cTWAS. Multigroup
cTWAS contain much more eGenes due to constructing a union.</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">MultiGroup</th>
<th align="right">SingleGrouop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0hr_GABA</td>
<td align="right">4157</td>
<td align="right">1798</td>
</tr>
<tr class="even">
<td align="left">1hr_GABA</td>
<td align="right">4183</td>
<td align="right">2182</td>
</tr>
<tr class="odd">
<td align="left">6hr_GABA</td>
<td align="right">4128</td>
<td align="right">2296</td>
</tr>
<tr class="even">
<td align="left">0hr_nmglut</td>
<td align="right">2554</td>
<td align="right">926</td>
</tr>
<tr class="odd">
<td align="left">1hr_nmglut</td>
<td align="right">2516</td>
<td align="right">1323</td>
</tr>
<tr class="even">
<td align="left">6hr_nmglut</td>
<td align="right">2562</td>
<td align="right">1186</td>
</tr>
<tr class="odd">
<td align="left">0hr_npglut</td>
<td align="right">5150</td>
<td align="right">2415</td>
</tr>
<tr class="even">
<td align="left">1hr_npglut</td>
<td align="right">5113</td>
<td align="right">2917</td>
</tr>
<tr class="odd">
<td align="left">6hr_npglut</td>
<td align="right">5170</td>
<td align="right">2976</td>
</tr>
</tbody>
</table>
</div>
<div id="initial-parameter-estimates" class="section level2">
<h2>Initial parameter estimates</h2>
<p>Estimating all parameters by sampling 10% of SNPs. (thin=0.1)</p>
<div id="ctwas-parameter-estimates-for-gaba" class="section level3">
<h3>cTWAS parameter estimates for GABA</h3>
<div class="figure">
<img src="assets/GABA.png" alt="" />
<p class="caption">GABA multigroup</p>
</div>
</div>
<div id="ctwas-parameter-estimates-for-nmglut" class="section level3">
<h3>cTWAS parameter estimates for nmglut</h3>
<div class="figure">
<img src="assets/nmglut.png" alt="" />
<p class="caption">nmglut multigroup</p>
</div>
</div>
<div id="ctwas-parameter-estimates-for-npglut" class="section level3">
<h3>cTWAS parameter estimates for npglut</h3>
<div class="figure">
<img src="assets/npglut.png" alt="" />
<p class="caption">npglut multigroup</p>
</div>
</div>
<div id="comparing-with-single-group-parameters" class="section level3">
<h3>Comparing with single group parameters</h3>
</div>
<div id="pve-proportion-for-each-cell-type" class="section level3">
<h3>PVE proportion for each cell type</h3>
<table>
<caption>Genes accounting for 4% to 6% of total explained
variance.</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">GABA</th>
<th align="left">npglut</th>
<th align="left">nmglut</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0 Hour</td>
<td align="left">1.83%</td>
<td align="left">2.16%</td>
<td align="left">0.18%</td>
</tr>
<tr class="even">
<td align="left">1 Hour</td>
<td align="left">1.54%</td>
<td align="left">1.22%</td>
<td align="left">2.37%</td>
</tr>
<tr class="odd">
<td align="left">6 Hour</td>
<td align="left">2.38%</td>
<td align="left">2.78%</td>
<td align="left">1.77%</td>
</tr>
<tr class="even">
<td align="left">Total</td>
<td align="left">5.75%</td>
<td align="left">6.15%</td>
<td align="left">4.32%</td>
</tr>
</tbody>
</table>
</div>
<div id="egenes-with-top-pips" class="section level3">
<h3>eGenes with top PIPs</h3>
<p>Posterior inclusion probability (PIP) across time points summed up to
be greater than 0.8 were considered top genes. GABA has 51 top genes.
NMglut has 38 top genes. NPglut has 41 top genes. These are the
candidate genes for running cTWAS with the full set of SNPs. “NA” means
that the gene have no siginificant eQTLs in the corresponding
condition.</p>
<table>
<caption>Top genes for the GABA cell type</caption>
<colgroup>
<col width="14%" />
<col width="14%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">PIP_sum</th>
<th align="right">PIP_0hr_GABA</th>
<th align="right">PIP_1hr_GABA</th>
<th align="right">PIP_6hr_GABA</th>
<th align="right">PIP_diff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NUDT22</td>
<td align="right">1.0869734</td>
<td align="right">0.9378652</td>
<td align="right">0.1491082</td>
<td align="right">NA</td>
<td align="right">-0.7887570</td>
</tr>
<tr class="even">
<td align="left">GALK2</td>
<td align="right">1.0499762</td>
<td align="right">0.0116613</td>
<td align="right">0.0472718</td>
<td align="right">0.9910431</td>
<td align="right">1.0266537</td>
</tr>
<tr class="odd">
<td align="left">ANKRD39</td>
<td align="right">1.0264764</td>
<td align="right">0.9809026</td>
<td align="right">0.0293496</td>
<td align="right">0.0162242</td>
<td align="right">-0.9353288</td>
</tr>
<tr class="even">
<td align="left">ARL14EP</td>
<td align="right">1.0162830</td>
<td align="right">0.0199778</td>
<td align="right">0.0147671</td>
<td align="right">0.9815381</td>
<td align="right">0.9763274</td>
</tr>
<tr class="odd">
<td align="left">CHAC2</td>
<td align="right">1.0125357</td>
<td align="right">0.7962035</td>
<td align="right">0.2030943</td>
<td align="right">0.0132379</td>
<td align="right">-0.5798714</td>
</tr>
<tr class="even">
<td align="left">HLA-DMA</td>
<td align="right">1.0105269</td>
<td align="right">0.2890632</td>
<td align="right">0.3452758</td>
<td align="right">0.3761879</td>
<td align="right">0.4324005</td>
</tr>
<tr class="odd">
<td align="left">CROT</td>
<td align="right">1.0057338</td>
<td align="right">0.0157423</td>
<td align="right">NA</td>
<td align="right">0.9899916</td>
<td align="right">0.9742493</td>
</tr>
<tr class="even">
<td align="left">EFHB</td>
<td align="right">1.0045376</td>
<td align="right">0.0242296</td>
<td align="right">0.9646853</td>
<td align="right">0.0156227</td>
<td align="right">0.9560784</td>
</tr>
<tr class="odd">
<td align="left">PCDH7</td>
<td align="right">1.0042423</td>
<td align="right">0.9798550</td>
<td align="right">0.0088368</td>
<td align="right">0.0155505</td>
<td align="right">-0.9554677</td>
</tr>
<tr class="even">
<td align="left">TMEM219</td>
<td align="right">1.0033574</td>
<td align="right">0.0298658</td>
<td align="right">NA</td>
<td align="right">0.9734916</td>
<td align="right">0.9436257</td>
</tr>
<tr class="odd">
<td align="left">SERPINI1</td>
<td align="right">0.9984995</td>
<td align="right">0.1716374</td>
<td align="right">0.3231191</td>
<td align="right">0.5037430</td>
<td align="right">0.6552247</td>
</tr>
<tr class="even">
<td align="left">GGNBP2</td>
<td align="right">0.9939430</td>
<td align="right">0.0244289</td>
<td align="right">0.8963715</td>
<td align="right">0.0731426</td>
<td align="right">0.9450851</td>
</tr>
<tr class="odd">
<td align="left">RASA3</td>
<td align="right">0.9938736</td>
<td align="right">0.9632367</td>
<td align="right">0.0104415</td>
<td align="right">0.0201953</td>
<td align="right">-0.9325999</td>
</tr>
<tr class="even">
<td align="left">RBBP5</td>
<td align="right">0.9926453</td>
<td align="right">0.0270800</td>
<td align="right">0.3773562</td>
<td align="right">0.5882090</td>
<td align="right">0.9384852</td>
</tr>
<tr class="odd">
<td align="left">PPFIA3</td>
<td align="right">0.9923421</td>
<td align="right">0.7739324</td>
<td align="right">0.1911953</td>
<td align="right">0.0272145</td>
<td align="right">-0.5555226</td>
</tr>
<tr class="even">
<td align="left">CDPF1</td>
<td align="right">0.9923198</td>
<td align="right">0.9651247</td>
<td align="right">0.0107282</td>
<td align="right">0.0164669</td>
<td align="right">-0.9379297</td>
</tr>
<tr class="odd">
<td align="left">GLTP</td>
<td align="right">0.9889936</td>
<td align="right">0.0221734</td>
<td align="right">0.0095464</td>
<td align="right">0.9572739</td>
<td align="right">0.9446469</td>
</tr>
<tr class="even">
<td align="left">MFAP3</td>
<td align="right">0.9862815</td>
<td align="right">0.0703852</td>
<td align="right">0.4511072</td>
<td align="right">0.4647891</td>
<td align="right">0.8455112</td>
</tr>
<tr class="odd">
<td align="left">RCC1</td>
<td align="right">0.9851396</td>
<td align="right">0.0198152</td>
<td align="right">0.9352244</td>
<td align="right">0.0301000</td>
<td align="right">0.9455092</td>
</tr>
<tr class="even">
<td align="left">NDUFA6</td>
<td align="right">0.9767506</td>
<td align="right">0.0074694</td>
<td align="right">0.9595206</td>
<td align="right">0.0097605</td>
<td align="right">0.9618117</td>
</tr>
<tr class="odd">
<td align="left">MCHR1</td>
<td align="right">0.9630899</td>
<td align="right">0.0066189</td>
<td align="right">0.0042811</td>
<td align="right">0.9521898</td>
<td align="right">0.9498521</td>
</tr>
<tr class="even">
<td align="left">LRCH4</td>
<td align="right">0.9616850</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.9616850</td>
<td align="right">0.9616850</td>
</tr>
<tr class="odd">
<td align="left">CEBPZ</td>
<td align="right">0.9601061</td>
<td align="right">0.0265127</td>
<td align="right">0.0164909</td>
<td align="right">0.9171025</td>
<td align="right">0.9070807</td>
</tr>
<tr class="even">
<td align="left">ATP5MC1</td>
<td align="right">0.9594618</td>
<td align="right">0.0701369</td>
<td align="right">0.3475172</td>
<td align="right">0.5418077</td>
<td align="right">0.8191881</td>
</tr>
<tr class="odd">
<td align="left">SYNGR1</td>
<td align="right">0.9535737</td>
<td align="right">0.9164597</td>
<td align="right">0.0188808</td>
<td align="right">0.0182332</td>
<td align="right">-0.8793458</td>
</tr>
<tr class="even">
<td align="left">GPR137</td>
<td align="right">0.9472145</td>
<td align="right">NA</td>
<td align="right">0.0065741</td>
<td align="right">0.9406404</td>
<td align="right">0.9472145</td>
</tr>
<tr class="odd">
<td align="left">XRN2</td>
<td align="right">0.9463454</td>
<td align="right">0.0158973</td>
<td align="right">0.0212032</td>
<td align="right">0.9092448</td>
<td align="right">0.9145507</td>
</tr>
<tr class="even">
<td align="left">AS3MT</td>
<td align="right">0.9446092</td>
<td align="right">0.3247822</td>
<td align="right">0.2710037</td>
<td align="right">0.3488233</td>
<td align="right">0.2950448</td>
</tr>
<tr class="odd">
<td align="left">TBC1D13</td>
<td align="right">0.9377078</td>
<td align="right">0.9022253</td>
<td align="right">0.0126538</td>
<td align="right">0.0228287</td>
<td align="right">-0.8667428</td>
</tr>
<tr class="even">
<td align="left">TRA2B</td>
<td align="right">0.8954401</td>
<td align="right">0.8352478</td>
<td align="right">NA</td>
<td align="right">0.0601924</td>
<td align="right">-0.7750554</td>
</tr>
<tr class="odd">
<td align="left">RBM6</td>
<td align="right">0.8934253</td>
<td align="right">0.0897532</td>
<td align="right">0.0121848</td>
<td align="right">0.7914873</td>
<td align="right">0.7139188</td>
</tr>
<tr class="even">
<td align="left">PCNX1</td>
<td align="right">0.8729807</td>
<td align="right">0.8729807</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">-0.8729807</td>
</tr>
<tr class="odd">
<td align="left">FBXO44</td>
<td align="right">0.8717883</td>
<td align="right">0.0132596</td>
<td align="right">NA</td>
<td align="right">0.8585287</td>
<td align="right">0.8452691</td>
</tr>
<tr class="even">
<td align="left">NPIPB2</td>
<td align="right">0.8691862</td>
<td align="right">0.3650181</td>
<td align="right">0.0290137</td>
<td align="right">0.4751544</td>
<td align="right">0.1391500</td>
</tr>
<tr class="odd">
<td align="left">LRMDA</td>
<td align="right">0.8646979</td>
<td align="right">0.0384553</td>
<td align="right">0.0102447</td>
<td align="right">0.8159978</td>
<td align="right">0.7877872</td>
</tr>
<tr class="even">
<td align="left">KCNMA1</td>
<td align="right">0.8631925</td>
<td align="right">0.1358983</td>
<td align="right">0.6672424</td>
<td align="right">0.0600518</td>
<td align="right">0.5913959</td>
</tr>
<tr class="odd">
<td align="left">PLAA</td>
<td align="right">0.8602175</td>
<td align="right">0.8097175</td>
<td align="right">0.0327660</td>
<td align="right">0.0177341</td>
<td align="right">-0.7592174</td>
</tr>
<tr class="even">
<td align="left">NLRC5</td>
<td align="right">0.8589574</td>
<td align="right">0.0694784</td>
<td align="right">0.7671048</td>
<td align="right">0.0223742</td>
<td align="right">0.7200007</td>
</tr>
<tr class="odd">
<td align="left">ZNF852</td>
<td align="right">0.8557082</td>
<td align="right">0.8367477</td>
<td align="right">0.0189604</td>
<td align="right">NA</td>
<td align="right">-0.8177873</td>
</tr>
<tr class="even">
<td align="left">CHL1</td>
<td align="right">0.8369448</td>
<td align="right">0.2667749</td>
<td align="right">0.2229172</td>
<td align="right">0.3472528</td>
<td align="right">0.3033951</td>
</tr>
<tr class="odd">
<td align="left">DCP1B</td>
<td align="right">0.8324851</td>
<td align="right">0.7979858</td>
<td align="right">0.0067389</td>
<td align="right">0.0277604</td>
<td align="right">-0.7634866</td>
</tr>
<tr class="even">
<td align="left">CTNNAL1</td>
<td align="right">0.8305312</td>
<td align="right">0.0148831</td>
<td align="right">0.8156481</td>
<td align="right">NA</td>
<td align="right">0.8007650</td>
</tr>
<tr class="odd">
<td align="left">INSYN2A</td>
<td align="right">0.8291783</td>
<td align="right">0.8291783</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">-0.8291783</td>
</tr>
<tr class="even">
<td align="left">ATG13</td>
<td align="right">0.8277512</td>
<td align="right">0.7796507</td>
<td align="right">0.0318826</td>
<td align="right">0.0162179</td>
<td align="right">-0.7315502</td>
</tr>
<tr class="odd">
<td align="left">CCDC60</td>
<td align="right">0.8207030</td>
<td align="right">0.7074528</td>
<td align="right">0.0387397</td>
<td align="right">0.0745105</td>
<td align="right">-0.5942026</td>
</tr>
<tr class="even">
<td align="left">TRIM33</td>
<td align="right">0.8182854</td>
<td align="right">0.0178044</td>
<td align="right">0.5991188</td>
<td align="right">0.2013622</td>
<td align="right">0.7826766</td>
</tr>
<tr class="odd">
<td align="left">FAM120AOS</td>
<td align="right">0.8148122</td>
<td align="right">0.0519945</td>
<td align="right">0.0434067</td>
<td align="right">0.7194110</td>
<td align="right">0.7108232</td>
</tr>
<tr class="even">
<td align="left">CDIP1</td>
<td align="right">0.8142464</td>
<td align="right">NA</td>
<td align="right">0.7703893</td>
<td align="right">0.0438571</td>
<td align="right">0.8142464</td>
</tr>
<tr class="odd">
<td align="left">ZNF365</td>
<td align="right">0.8070291</td>
<td align="right">0.0593247</td>
<td align="right">0.3079213</td>
<td align="right">0.4397831</td>
<td align="right">0.6883798</td>
</tr>
<tr class="even">
<td align="left">SCML4</td>
<td align="right">0.8053856</td>
<td align="right">0.0198502</td>
<td align="right">0.7855354</td>
<td align="right">NA</td>
<td align="right">0.7656851</td>
</tr>
<tr class="odd">
<td align="left">MAP2K5</td>
<td align="right">0.8026013</td>
<td align="right">0.0146209</td>
<td align="right">0.7684403</td>
<td align="right">0.0195401</td>
<td align="right">0.7733594</td>
</tr>
</tbody>
</table>
<table>
<caption>Top genes for the nmglut cell type</caption>
<colgroup>
<col width="12%" />
<col width="13%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">PIP_sum</th>
<th align="right">PIP_0hr_nmglut</th>
<th align="right">PIP_1hr_nmglut</th>
<th align="right">PIP_6hr_nmglut</th>
<th align="right">PIP_diff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">KIAA0319</td>
<td align="right">1.5386851</td>
<td align="right">NA</td>
<td align="right">0.9290925</td>
<td align="right">0.6095926</td>
<td align="right">1.5386851</td>
</tr>
<tr class="even">
<td align="left">TDRD6</td>
<td align="right">1.0918803</td>
<td align="right">0.0023007</td>
<td align="right">0.9485020</td>
<td align="right">0.1410775</td>
<td align="right">1.0872789</td>
</tr>
<tr class="odd">
<td align="left">NGEF</td>
<td align="right">1.0546695</td>
<td align="right">0.0021949</td>
<td align="right">0.9540519</td>
<td align="right">0.0984227</td>
<td align="right">1.0502797</td>
</tr>
<tr class="even">
<td align="left">GLIPR2</td>
<td align="right">1.0352918</td>
<td align="right">0.0080940</td>
<td align="right">0.9850915</td>
<td align="right">0.0421063</td>
<td align="right">1.0191039</td>
</tr>
<tr class="odd">
<td align="left">HLA-DMA</td>
<td align="right">1.0340127</td>
<td align="right">0.0064863</td>
<td align="right">0.0843759</td>
<td align="right">0.9431505</td>
<td align="right">1.0210401</td>
</tr>
<tr class="even">
<td align="left">ZMYND10</td>
<td align="right">1.0304008</td>
<td align="right">NA</td>
<td align="right">0.0360545</td>
<td align="right">0.9943463</td>
<td align="right">1.0304008</td>
</tr>
<tr class="odd">
<td align="left">FOXN2</td>
<td align="right">1.0254627</td>
<td align="right">0.0069370</td>
<td align="right">0.9888052</td>
<td align="right">0.0297205</td>
<td align="right">1.0115887</td>
</tr>
<tr class="even">
<td align="left">ANKRD45</td>
<td align="right">1.0128922</td>
<td align="right">NA</td>
<td align="right">0.9604396</td>
<td align="right">0.0524525</td>
<td align="right">1.0128922</td>
</tr>
<tr class="odd">
<td align="left">C17orf75</td>
<td align="right">1.0108147</td>
<td align="right">NA</td>
<td align="right">0.1619486</td>
<td align="right">0.8488661</td>
<td align="right">1.0108147</td>
</tr>
<tr class="even">
<td align="left">SERPINI1</td>
<td align="right">1.0049270</td>
<td align="right">0.0028291</td>
<td align="right">0.7409321</td>
<td align="right">0.2611659</td>
<td align="right">0.9992688</td>
</tr>
<tr class="odd">
<td align="left">CYP7B1</td>
<td align="right">1.0040565</td>
<td align="right">0.0026529</td>
<td align="right">0.0363761</td>
<td align="right">0.9650274</td>
<td align="right">0.9987506</td>
</tr>
<tr class="even">
<td align="left">CPT1C</td>
<td align="right">0.9879569</td>
<td align="right">NA</td>
<td align="right">0.9879569</td>
<td align="right">NA</td>
<td align="right">0.9879569</td>
</tr>
<tr class="odd">
<td align="left">DNPH1</td>
<td align="right">0.9806041</td>
<td align="right">0.0154048</td>
<td align="right">0.9386044</td>
<td align="right">0.0265949</td>
<td align="right">0.9497946</td>
</tr>
<tr class="even">
<td align="left">DCP1B</td>
<td align="right">0.9670935</td>
<td align="right">0.0147349</td>
<td align="right">0.9169144</td>
<td align="right">0.0354442</td>
<td align="right">0.9376237</td>
</tr>
<tr class="odd">
<td align="left">ACYP2</td>
<td align="right">0.9551205</td>
<td align="right">0.0032281</td>
<td align="right">0.0597019</td>
<td align="right">0.8921905</td>
<td align="right">0.9486643</td>
</tr>
<tr class="even">
<td align="left">AS3MT</td>
<td align="right">0.9540593</td>
<td align="right">0.0018455</td>
<td align="right">0.6258593</td>
<td align="right">0.3263545</td>
<td align="right">0.9503682</td>
</tr>
<tr class="odd">
<td align="left">NPIPB2</td>
<td align="right">0.9490831</td>
<td align="right">0.0482533</td>
<td align="right">0.6238749</td>
<td align="right">0.2769549</td>
<td align="right">0.8525765</td>
</tr>
<tr class="even">
<td align="left">OXLD1</td>
<td align="right">0.9457363</td>
<td align="right">0.0027329</td>
<td align="right">0.0460262</td>
<td align="right">0.8969771</td>
<td align="right">0.9402705</td>
</tr>
<tr class="odd">
<td align="left">ATP5PO</td>
<td align="right">0.9380107</td>
<td align="right">0.0672226</td>
<td align="right">0.8707881</td>
<td align="right">NA</td>
<td align="right">0.8035655</td>
</tr>
<tr class="even">
<td align="left">CXXC5</td>
<td align="right">0.9255876</td>
<td align="right">0.0018701</td>
<td align="right">0.0253466</td>
<td align="right">0.8983708</td>
<td align="right">0.9218473</td>
</tr>
<tr class="odd">
<td align="left">CYP2D6</td>
<td align="right">0.9214084</td>
<td align="right">0.0017077</td>
<td align="right">0.8428334</td>
<td align="right">0.0768672</td>
<td align="right">0.9179929</td>
</tr>
<tr class="even">
<td align="left">CCDC122</td>
<td align="right">0.9125166</td>
<td align="right">0.0062697</td>
<td align="right">0.3283876</td>
<td align="right">0.5778593</td>
<td align="right">0.8999772</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS2</td>
<td align="right">0.8970327</td>
<td align="right">0.0033236</td>
<td align="right">0.8683194</td>
<td align="right">0.0253897</td>
<td align="right">0.8903855</td>
</tr>
<tr class="even">
<td align="left">CD46</td>
<td align="right">0.8827728</td>
<td align="right">0.0528717</td>
<td align="right">0.0409581</td>
<td align="right">0.7889430</td>
<td align="right">0.7770295</td>
</tr>
<tr class="odd">
<td align="left">ATE1</td>
<td align="right">0.8704377</td>
<td align="right">0.0057718</td>
<td align="right">0.4963777</td>
<td align="right">0.3682882</td>
<td align="right">0.8588941</td>
</tr>
<tr class="even">
<td align="left">KALRN</td>
<td align="right">0.8671839</td>
<td align="right">0.0038857</td>
<td align="right">0.0491531</td>
<td align="right">0.8141450</td>
<td align="right">0.8594124</td>
</tr>
<tr class="odd">
<td align="left">DNAH7</td>
<td align="right">0.8571767</td>
<td align="right">0.0041214</td>
<td align="right">0.0409945</td>
<td align="right">0.8120609</td>
<td align="right">0.8489340</td>
</tr>
<tr class="even">
<td align="left">TSPAN2</td>
<td align="right">0.8556092</td>
<td align="right">0.0032428</td>
<td align="right">0.0349294</td>
<td align="right">0.8174370</td>
<td align="right">0.8491236</td>
</tr>
<tr class="odd">
<td align="left">RFLNA</td>
<td align="right">0.8553334</td>
<td align="right">0.7135342</td>
<td align="right">0.0417782</td>
<td align="right">0.1000210</td>
<td align="right">-0.5717351</td>
</tr>
<tr class="even">
<td align="left">EEFSEC</td>
<td align="right">0.8534798</td>
<td align="right">0.0502977</td>
<td align="right">0.4707613</td>
<td align="right">0.3324208</td>
<td align="right">0.7528844</td>
</tr>
<tr class="odd">
<td align="left">KDM4C</td>
<td align="right">0.8516559</td>
<td align="right">0.0044345</td>
<td align="right">0.8165295</td>
<td align="right">0.0306920</td>
<td align="right">0.8427870</td>
</tr>
<tr class="even">
<td align="left">SMYD5</td>
<td align="right">0.8471234</td>
<td align="right">0.0039728</td>
<td align="right">0.8180220</td>
<td align="right">0.0251286</td>
<td align="right">0.8391778</td>
</tr>
<tr class="odd">
<td align="left">METRN</td>
<td align="right">0.8431658</td>
<td align="right">0.0147437</td>
<td align="right">0.1841257</td>
<td align="right">0.6442964</td>
<td align="right">0.8136784</td>
</tr>
<tr class="even">
<td align="left">SNX19</td>
<td align="right">0.8269636</td>
<td align="right">0.0351221</td>
<td align="right">0.4548956</td>
<td align="right">0.3369459</td>
<td align="right">0.7567194</td>
</tr>
<tr class="odd">
<td align="left">C9orf72</td>
<td align="right">0.8206724</td>
<td align="right">0.0024250</td>
<td align="right">0.1023970</td>
<td align="right">0.7158504</td>
<td align="right">0.8158224</td>
</tr>
<tr class="even">
<td align="left">TNR</td>
<td align="right">0.8139383</td>
<td align="right">0.0023089</td>
<td align="right">0.0232147</td>
<td align="right">0.7884147</td>
<td align="right">0.8093206</td>
</tr>
<tr class="odd">
<td align="left">RCBTB1</td>
<td align="right">0.8095863</td>
<td align="right">0.0723736</td>
<td align="right">0.0481942</td>
<td align="right">0.6890185</td>
<td align="right">0.6648391</td>
</tr>
<tr class="even">
<td align="left">SPATS2</td>
<td align="right">0.8081127</td>
<td align="right">NA</td>
<td align="right">0.8081127</td>
<td align="right">NA</td>
<td align="right">0.8081127</td>
</tr>
</tbody>
</table>
<table>
<caption>Top genes for the nmglut cell type</caption>
<colgroup>
<col width="12%" />
<col width="13%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">PIP_sum</th>
<th align="right">PIP_0hr_npglut</th>
<th align="right">PIP_1hr_npglut</th>
<th align="right">PIP_6hr_npglut</th>
<th align="right">PIP_diff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AUTS2</td>
<td align="right">1.2161510</td>
<td align="right">0.1644127</td>
<td align="right">0.7695881</td>
<td align="right">0.2821502</td>
<td align="right">0.8873256</td>
</tr>
<tr class="even">
<td align="left">ZNF823</td>
<td align="right">1.0427663</td>
<td align="right">0.0385086</td>
<td align="right">0.9943159</td>
<td align="right">0.0099418</td>
<td align="right">0.9657491</td>
</tr>
<tr class="odd">
<td align="left">BMF</td>
<td align="right">1.0210662</td>
<td align="right">0.9908971</td>
<td align="right">0.0160277</td>
<td align="right">0.0141415</td>
<td align="right">-0.9607279</td>
</tr>
<tr class="even">
<td align="left">THBS3</td>
<td align="right">1.0061874</td>
<td align="right">0.0122685</td>
<td align="right">NA</td>
<td align="right">0.9939189</td>
<td align="right">0.9816504</td>
</tr>
<tr class="odd">
<td align="left">CRTAP</td>
<td align="right">1.0038560</td>
<td align="right">0.0317668</td>
<td align="right">0.2055465</td>
<td align="right">0.7665427</td>
<td align="right">0.9403224</td>
</tr>
<tr class="even">
<td align="left">CRAT</td>
<td align="right">1.0031045</td>
<td align="right">0.9589551</td>
<td align="right">0.0079337</td>
<td align="right">0.0362157</td>
<td align="right">-0.9148056</td>
</tr>
<tr class="odd">
<td align="left">NAGA</td>
<td align="right">1.0012158</td>
<td align="right">0.0049169</td>
<td align="right">0.0028616</td>
<td align="right">0.9934373</td>
<td align="right">0.9913820</td>
</tr>
<tr class="even">
<td align="left">TMX1</td>
<td align="right">0.9860060</td>
<td align="right">0.0119220</td>
<td align="right">0.0064451</td>
<td align="right">0.9676389</td>
<td align="right">0.9621620</td>
</tr>
<tr class="odd">
<td align="left">SMDT1</td>
<td align="right">0.9858649</td>
<td align="right">0.0057753</td>
<td align="right">0.4699751</td>
<td align="right">0.5101145</td>
<td align="right">0.9743143</td>
</tr>
<tr class="even">
<td align="left">OLA1</td>
<td align="right">0.9804494</td>
<td align="right">0.0078095</td>
<td align="right">NA</td>
<td align="right">0.9726399</td>
<td align="right">0.9648304</td>
</tr>
<tr class="odd">
<td align="left">MOB3A</td>
<td align="right">0.9769611</td>
<td align="right">0.9601726</td>
<td align="right">NA</td>
<td align="right">0.0167885</td>
<td align="right">-0.9433841</td>
</tr>
<tr class="even">
<td align="left">GCH1</td>
<td align="right">0.9712662</td>
<td align="right">0.0155235</td>
<td align="right">0.0086427</td>
<td align="right">0.9471000</td>
<td align="right">0.9402192</td>
</tr>
<tr class="odd">
<td align="left">CD46</td>
<td align="right">0.9675918</td>
<td align="right">0.2290288</td>
<td align="right">0.2187498</td>
<td align="right">0.5198132</td>
<td align="right">0.5095342</td>
</tr>
<tr class="even">
<td align="left">GIGYF1</td>
<td align="right">0.9367359</td>
<td align="right">0.9225593</td>
<td align="right">NA</td>
<td align="right">0.0141766</td>
<td align="right">-0.9083827</td>
</tr>
<tr class="odd">
<td align="left">TIGD1</td>
<td align="right">0.9315737</td>
<td align="right">0.0244331</td>
<td align="right">0.0086842</td>
<td align="right">0.8984565</td>
<td align="right">0.8827076</td>
</tr>
<tr class="even">
<td align="left">MPHOSPH9</td>
<td align="right">0.9245093</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.9245093</td>
<td align="right">0.9245093</td>
</tr>
<tr class="odd">
<td align="left">SLC25A12</td>
<td align="right">0.9235158</td>
<td align="right">0.0118783</td>
<td align="right">0.8934398</td>
<td align="right">0.0181977</td>
<td align="right">0.8997592</td>
</tr>
<tr class="even">
<td align="left">PPP2R3C</td>
<td align="right">0.9165499</td>
<td align="right">0.0432124</td>
<td align="right">0.8570450</td>
<td align="right">0.0162925</td>
<td align="right">0.8301251</td>
</tr>
<tr class="odd">
<td align="left">ATP5MC1</td>
<td align="right">0.9130450</td>
<td align="right">0.0956909</td>
<td align="right">0.0100586</td>
<td align="right">0.8072954</td>
<td align="right">0.7216632</td>
</tr>
<tr class="even">
<td align="left">DISP3</td>
<td align="right">0.9081443</td>
<td align="right">0.0144512</td>
<td align="right">0.2760225</td>
<td align="right">0.6176706</td>
<td align="right">0.8792419</td>
</tr>
<tr class="odd">
<td align="left">NGEF</td>
<td align="right">0.9058406</td>
<td align="right">0.0101174</td>
<td align="right">0.8273583</td>
<td align="right">0.0683649</td>
<td align="right">0.8856057</td>
</tr>
<tr class="even">
<td align="left">MOCS3</td>
<td align="right">0.9018225</td>
<td align="right">0.1050444</td>
<td align="right">0.1620302</td>
<td align="right">0.6347479</td>
<td align="right">0.6917337</td>
</tr>
<tr class="odd">
<td align="left">TNFRSF19</td>
<td align="right">0.9018144</td>
<td align="right">0.8370134</td>
<td align="right">0.0069503</td>
<td align="right">0.0578506</td>
<td align="right">-0.7722125</td>
</tr>
<tr class="even">
<td align="left">ITSN1</td>
<td align="right">0.8888481</td>
<td align="right">0.0320017</td>
<td align="right">0.2364369</td>
<td align="right">0.6204095</td>
<td align="right">0.8248447</td>
</tr>
<tr class="odd">
<td align="left">ARHGEF7</td>
<td align="right">0.8873169</td>
<td align="right">0.0264778</td>
<td align="right">0.0062626</td>
<td align="right">0.8545766</td>
<td align="right">0.8343614</td>
</tr>
<tr class="even">
<td align="left">PCYT1A</td>
<td align="right">0.8869254</td>
<td align="right">NA</td>
<td align="right">0.0122529</td>
<td align="right">0.8746725</td>
<td align="right">0.8869254</td>
</tr>
<tr class="odd">
<td align="left">CHRNA3</td>
<td align="right">0.8809802</td>
<td align="right">0.7658896</td>
<td align="right">0.0716908</td>
<td align="right">0.0433998</td>
<td align="right">-0.6507990</td>
</tr>
<tr class="even">
<td align="left">APC2</td>
<td align="right">0.8788322</td>
<td align="right">0.5110025</td>
<td align="right">0.2919341</td>
<td align="right">0.0758957</td>
<td align="right">-0.1431728</td>
</tr>
<tr class="odd">
<td align="left">CHAC2</td>
<td align="right">0.8747098</td>
<td align="right">0.8668216</td>
<td align="right">0.0078883</td>
<td align="right">NA</td>
<td align="right">-0.8589333</td>
</tr>
<tr class="even">
<td align="left">ABITRAM</td>
<td align="right">0.8744183</td>
<td align="right">0.0154501</td>
<td align="right">0.8589683</td>
<td align="right">NA</td>
<td align="right">0.8435182</td>
</tr>
<tr class="odd">
<td align="left">HECW2</td>
<td align="right">0.8734191</td>
<td align="right">0.8247486</td>
<td align="right">0.0370050</td>
<td align="right">0.0116655</td>
<td align="right">-0.7760781</td>
</tr>
<tr class="even">
<td align="left">ARPP19</td>
<td align="right">0.8706664</td>
<td align="right">0.1840912</td>
<td align="right">0.6530843</td>
<td align="right">0.0334909</td>
<td align="right">0.5024839</td>
</tr>
<tr class="odd">
<td align="left">CSNK2B</td>
<td align="right">0.8696222</td>
<td align="right">0.0000161</td>
<td align="right">0.0000044</td>
<td align="right">0.8696018</td>
<td align="right">0.8695901</td>
</tr>
<tr class="even">
<td align="left">TDRD6</td>
<td align="right">0.8675521</td>
<td align="right">0.3682195</td>
<td align="right">0.0293193</td>
<td align="right">0.4700133</td>
<td align="right">0.1311130</td>
</tr>
<tr class="odd">
<td align="left">ZBED4</td>
<td align="right">0.8649705</td>
<td align="right">0.2616165</td>
<td align="right">0.2549961</td>
<td align="right">0.3483579</td>
<td align="right">0.3417375</td>
</tr>
<tr class="even">
<td align="left">ALMS1</td>
<td align="right">0.8612558</td>
<td align="right">0.0195807</td>
<td align="right">0.8167506</td>
<td align="right">0.0249245</td>
<td align="right">0.8220944</td>
</tr>
<tr class="odd">
<td align="left">DNAH7</td>
<td align="right">0.8322135</td>
<td align="right">0.3629555</td>
<td align="right">0.0057035</td>
<td align="right">0.4635545</td>
<td align="right">0.1063025</td>
</tr>
<tr class="even">
<td align="left">MAU2</td>
<td align="right">0.8224328</td>
<td align="right">0.0120092</td>
<td align="right">0.7887235</td>
<td align="right">0.0217001</td>
<td align="right">0.7984144</td>
</tr>
<tr class="odd">
<td align="left">HLA-DMA</td>
<td align="right">0.8195709</td>
<td align="right">0.3568805</td>
<td align="right">0.1431824</td>
<td align="right">0.3195079</td>
<td align="right">0.1058099</td>
</tr>
<tr class="even">
<td align="left">NPIPB2</td>
<td align="right">0.8118282</td>
<td align="right">0.2497166</td>
<td align="right">0.2435313</td>
<td align="right">0.3185803</td>
<td align="right">0.3123949</td>
</tr>
<tr class="odd">
<td align="left">ABHD17C</td>
<td align="right">0.8117265</td>
<td align="right">NA</td>
<td align="right">0.0276673</td>
<td align="right">0.7840591</td>
<td align="right">0.8117265</td>
</tr>
</tbody>
</table>
</div>
<div id="correlation-of-pip-across-cell-types" class="section level3">
<h3>Correlation of PIP across cell types</h3>
<p><img src="figure/cTWAS.Rmd/unnamed-chunk-6-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/neuron_stimulation_100line_eQTL/blob/8c088c5eeae82acf35d2f9089e4c0f13073a5509/docs/figure/cTWAS.Rmd/unnamed-chunk-6-1.png" target="_blank">8c088c5</a>
</td>
<td>
Lifan Liang
</td>
<td>
2023-10-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ocurrence-of-region-merging" class="section level3">
<h3>Ocurrence of region merging</h3>
<p>cTWAS will automatically merge LD regions if a gene spans across
multiple regions. This is computationally expensive. Fortunately, with
the merging option turned on, no merging was performed. This is probably
because none of the genes selected for multi-group cTWAS spanned
multiple LD regions.</p>
</div>
</div>
<div id="rerunning-ctwas-on-subset-fo-genes" class="section level2">
<h2>Rerunning cTWAS on subset fo genes</h2>
<p>Since PIP was estimated with 10% thining, we need to esstimate the
PIP and performed fine mapping on the top genes. Specifically, with the
same imputed gene expression and group priors, we run cTWAS at the
specific gene locations with the full set of SNPs.</p>
<div class="figure">
<img src="assets/rerun_vs_init.png" alt="" />
<p class="caption">PIP comparison between initial run and the rerun</p>
</div>
<p>Most top gene PIPs were overestimated in the initial run. There are
17 genes with PIP sums greater than 0.8 for the GABA cell type. Only 2
genes cross the threshold in both Glut cell types. In addition, all the
remaining top genes in the Glut cell types were also in the top gene
list of GABA. The NPIPB2 has high PIP in all cell types.</p>
<table>
<colgroup>
<col width="13%" />
<col width="14%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">PIP_sum</th>
<th align="right">PIP_0hr_GABA</th>
<th align="right">PIP_1hr_GABA</th>
<th align="right">PIP_6hr_GABA</th>
<th align="right">PIP_diff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GALK2</td>
<td align="right">1.0321472</td>
<td align="right">0.0211655</td>
<td align="right">0.0518846</td>
<td align="right">0.9590971</td>
<td align="right">0.9898162</td>
</tr>
<tr class="even">
<td align="left">CDPF1</td>
<td align="right">0.9777592</td>
<td align="right">0.9365297</td>
<td align="right">0.0159119</td>
<td align="right">0.0253175</td>
<td align="right">-0.8953003</td>
</tr>
<tr class="odd">
<td align="left">SERPINI1</td>
<td align="right">0.9583311</td>
<td align="right">0.1762213</td>
<td align="right">0.3060035</td>
<td align="right">0.4761063</td>
<td align="right">0.6058885</td>
</tr>
<tr class="even">
<td align="left">ARL14EP</td>
<td align="right">0.9400219</td>
<td align="right">0.0265764</td>
<td align="right">0.0192168</td>
<td align="right">0.8942288</td>
<td align="right">0.8868692</td>
</tr>
<tr class="odd">
<td align="left">CROT</td>
<td align="right">0.9242892</td>
<td align="right">0.0094062</td>
<td align="right">NA</td>
<td align="right">0.9148831</td>
<td align="right">0.9054769</td>
</tr>
<tr class="even">
<td align="left">EFHB</td>
<td align="right">0.8881553</td>
<td align="right">0.0220144</td>
<td align="right">0.8464022</td>
<td align="right">0.0197388</td>
<td align="right">0.8441266</td>
</tr>
<tr class="odd">
<td align="left">DCP1B</td>
<td align="right">0.8839398</td>
<td align="right">0.8685502</td>
<td align="right">0.0033330</td>
<td align="right">0.0120566</td>
<td align="right">-0.8531606</td>
</tr>
<tr class="even">
<td align="left">PCDH7</td>
<td align="right">0.8681452</td>
<td align="right">0.8424954</td>
<td align="right">0.0092044</td>
<td align="right">0.0164454</td>
<td align="right">-0.8168456</td>
</tr>
<tr class="odd">
<td align="left">NPIPB2</td>
<td align="right">0.8554143</td>
<td align="right">0.3588716</td>
<td align="right">0.0296929</td>
<td align="right">0.4668498</td>
<td align="right">0.1376710</td>
</tr>
<tr class="even">
<td align="left">TMEM219</td>
<td align="right">0.8407320</td>
<td align="right">0.0402824</td>
<td align="right">NA</td>
<td align="right">0.8004497</td>
<td align="right">0.7601673</td>
</tr>
<tr class="odd">
<td align="left">ANKRD39</td>
<td align="right">0.8390485</td>
<td align="right">0.7881384</td>
<td align="right">0.0296520</td>
<td align="right">0.0212581</td>
<td align="right">-0.7372283</td>
</tr>
<tr class="even">
<td align="left">MCHR1</td>
<td align="right">0.8377931</td>
<td align="right">0.0017432</td>
<td align="right">0.0013532</td>
<td align="right">0.8346967</td>
<td align="right">0.8343067</td>
</tr>
<tr class="odd">
<td align="left">ATP5MC1</td>
<td align="right">0.8291203</td>
<td align="right">0.0630461</td>
<td align="right">0.2994933</td>
<td align="right">0.4665810</td>
<td align="right">0.7030282</td>
</tr>
<tr class="even">
<td align="left">HLA-DMA</td>
<td align="right">0.8179896</td>
<td align="right">0.2235701</td>
<td align="right">0.3021008</td>
<td align="right">0.2923187</td>
<td align="right">0.3708494</td>
</tr>
<tr class="odd">
<td align="left">RBBP5</td>
<td align="right">0.8128986</td>
<td align="right">0.0258590</td>
<td align="right">0.3080663</td>
<td align="right">0.4789733</td>
<td align="right">0.7611806</td>
</tr>
<tr class="even">
<td align="left">NDUFA6</td>
<td align="right">0.8116531</td>
<td align="right">0.0046170</td>
<td align="right">0.8010000</td>
<td align="right">0.0060361</td>
<td align="right">0.8024190</td>
</tr>
<tr class="odd">
<td align="left">RASA3</td>
<td align="right">0.8016648</td>
<td align="right">0.7550925</td>
<td align="right">0.0152470</td>
<td align="right">0.0313253</td>
<td align="right">-0.7085201</td>
</tr>
</tbody>
</table>
<table style="width:100%;">
<colgroup>
<col width="12%" />
<col width="13%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">PIP_sum</th>
<th align="right">PIP_0hr_nmglut</th>
<th align="right">PIP_1hr_nmglut</th>
<th align="right">PIP_6hr_nmglut</th>
<th align="right">PIP_diff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SERPINI1</td>
<td align="right">0.9837154</td>
<td align="right">0.0037831</td>
<td align="right">0.7221600</td>
<td align="right">0.2577723</td>
<td align="right">0.9761493</td>
</tr>
<tr class="even">
<td align="left">NPIPB2</td>
<td align="right">0.9285035</td>
<td align="right">0.0476253</td>
<td align="right">0.6109542</td>
<td align="right">0.2699240</td>
<td align="right">0.8332529</td>
</tr>
</tbody>
</table>
<table style="width:100%;">
<colgroup>
<col width="12%" />
<col width="13%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">PIP_sum</th>
<th align="right">PIP_0hr_npglut</th>
<th align="right">PIP_1hr_npglut</th>
<th align="right">PIP_6hr_npglut</th>
<th align="right">PIP_diff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NPIPB2</td>
<td align="right">0.8108719</td>
<td align="right">0.2481337</td>
<td align="right">0.2462959</td>
<td align="right">0.3164423</td>
<td align="right">0.3146044</td>
</tr>
<tr class="even">
<td align="left">TMEM219</td>
<td align="right">0.8060793</td>
<td align="right">0.0160886</td>
<td align="right">0.0091464</td>
<td align="right">0.7808443</td>
<td align="right">0.7739022</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.5   workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1 xfun_0.29        bslib_0.3.1      purrr_0.3.4     
 [5] colorspace_2.0-2 vctrs_0.3.8      generics_0.1.2   htmltools_0.5.2 
 [9] yaml_2.2.2       utf8_1.2.2       rlang_1.0.0      jquerylib_0.1.4 
[13] later_1.3.0      pillar_1.7.0     withr_2.4.3      glue_1.6.1      
[17] DBI_1.1.2        lifecycle_1.0.1  stringr_1.4.0    munsell_0.5.0   
[21] gtable_0.3.0     evaluate_0.14    knitr_1.37       callr_3.7.0     
[25] fastmap_1.1.0    httpuv_1.6.5     ps_1.6.0         fansi_1.0.2     
[29] highr_0.9        Rcpp_1.0.11      promises_1.2.0.1 scales_1.1.1    
[33] jsonlite_1.7.3   fs_1.5.2         digest_0.6.29    stringi_1.7.6   
[37] processx_3.5.2   dplyr_1.0.7      getPass_0.2-2    cowplot_1.1.1   
[41] rprojroot_2.0.2  grid_4.1.2       cli_3.1.1        tools_4.1.2     
[45] magrittr_2.0.2   sass_0.4.1       tibble_3.1.6     crayon_1.4.2    
[49] whisker_0.4      pkgconfig_2.0.3  ellipsis_0.3.2   assertthat_0.2.1
[53] rmarkdown_2.11   httr_1.4.2       rstudioapi_0.13  R6_2.5.1        
[57] git2r_0.29.0     compiler_4.1.2  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
